### PR TITLE
Added ts-auto-mock

### PR DESCRIPTION
**Pull request by haubitzer**
* refactored tasks.test to make reuse easier

### CODE REVIEW

**Comment by haubitzer** [comment]
Minor changes need after #94, So this PR will wait.

File: package.json [hunk 1]
```diff
@@ -66,5 +69,20 @@
...
+    "testEnvironment": "node",
+    "setupFiles": [
+      "<rootDir>config.ts"
+    ],
+    "transform": {
+      ".(ts|tsx)": "ts-jest"
+    },
+    "globals": {
+      "ts-jest": {
+        "compiler": "ttypescript"
```
Comment (benj56) [inline package.json:84]: Looks like a typo but not sure:
```suggestion
        "compiler": "typescript"
```

**Review by benj56 — DISMISSED** [review]
Looks useful :+1:

File: package.json (hunk 1 above)
Comment (haubitzer) [reply package.json:85]: it's not, https://typescript-tdd.github.io/ts-auto-mock/installation

File: src/tasks/tasks.test.ts [hunk 2]
```diff
@@ -1,104 +1,170 @@
...
-    });
+  test('with status 200 and Success false', async () => {
+    const mock = new AxiosMockAdapter(axios);
+    const record = createMock<Resource>();
+    const apiResponse = { Success: false, NumResults: 1, Results: [record] };
+
+    mock.onGet(readPath).reply(200, apiResponse);
+    const actual: Promise<Resource[]> = endpoint.read();
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return
+    expect(await actual.catch((err) => err)).toStrictEqual(apiResponse);
```
Comment (benj56) [inline src/tasks/tasks.test.ts:55]:
```suggestion
    expect(await actual.catch((err: typeof apiResponse) => err)).toStrictEqual(apiResponse);
```

**Review by benj56 — APPROVED** [review]

**Review by benj56 — APPROVED** [review]

**Pull request merged**